packages/core/src/browser/tree/tree-widget.tsx

```diff
@@ -571,10 +576,25 @@ export class TreeWidget extends ReactWidget implements StatefulWidget {
                 const { tooltip } = decoration;
                 const { data, fontData } = decoration as TreeDecoration.TailDecoration;
                 const color = (decoration as TreeDecoration.TailDecorationIcon).color;
-                const icon = (decoration as TreeDecoration.TailDecorationIcon).icon || (decoration as TreeDecoration.TailDecorationIconClass).iconClass;
-                const className = [TREE_NODE_SEGMENT_CLASS, TREE_NODE_TAIL_CLASS].join(' ');
+                const iconClass = (decoration as TreeDecoration.TailDecorationIconClass).iconClass;
```

Comment:
It seems we do not handle the `TreeDecoration.TailDecorationIcon` anymore.

Reply:
fixed with latest commit

Reply:
> Instead of introducing yet another interface, can we make TailDecorationIconClass and TailDecorationIcon optionally "clickable"?

I started this way, but then I learned that for command with icon, a dynamic css class is created. So the semantic of the iconClass is different. in the TailDecorationIcon and TailDecorationIconClass the class is FA based and in the command case, it is dynamic pointing to the url provided by the extension. Therefore I switched to own interface.

Reply:
> I have tried your changes and your extension in Gitpod, although it seemed to work, I could see the following error in the logs:

can you instruct me how to do the same? so I can try to debug it in the same environment - in gitpod?

Reply:
- I have opened your PR in Gitpod: https://gitpod.io/#https://github.com/theia-ide/theia/pull/4969
 - Create the `plugins` folder and clone your example there: `mkdir plugins && cd plugins && git clone https://github.com/ariel-bentu/theia-vscode-gaps`
 - Add `"skipLibCheck": true` to the `tsconfig.json` and run `npm i && npm run compile` for your extension.
 - Start Theia: `cd ../../ && yarn --cwd ./examples/browser start`.
 - Click on `Open Browser`.

![Screen Shot 2019-04-24 at 08 17 46](https://user-images.githubusercontent.com/1405703/56637016-dc4b4e00-666a-11e9-88c1-37b55297e985.jpg)

 - You can see the contributions; no errors so far.
 - Refersh your browser <kbd>Ctrl/Cmd</kbd>+<kbd>R</kbd>. If asked `Reload site?`, then `Reload`.
 - Now you can see the error, but the decorator icons are visible: `GET https://3000-bd0da029-e6dc-46dd-9902-0785ae55bea4.ws-eu0.gitpod.io/workspace/theia/plugins/theia-vscode-gaps/resources/light/bugs.svg 404`

Reply:
> I learned that for command with icon, a dynamic css class is created

I do not have much experience with VS Code extensions; are you saying the `TailDecorationCommand` interface can be used with VS Code extensions only?

Reply:
Ok, problem was that I used vsix of my example and you used it extracted (git clone), apparently my problematic logic (as pointed out in the pr) to convert absolute url to relative to hostedPlugin/plugin_id/... broke. I now added to the view meta data also the absolute extension path so the logic is now robust and makes no assumption.

A  for using it not in vscode extension, I suppose you mean theia extension? same way plugin-ext can use it, others can too.